projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cds/profile-tag.e2e-spec.ts

```diff
@@ -0,0 +1,44 @@
+import { clickAllowAllFromBanner } from '../../../helpers/anonymous-consents';
+import { Navigation } from '../../../helpers/navigation';
+import { CdsHelper } from '../../../helpers/vendor/cds/cds';
+import * as MercahndisingCarousel from '../../../helpers/vendor/cds/merchandising-carousel';
+import { ProfileTagHelper } from '../../../helpers/vendor/cds/profile-tag';
+describe('Profile-tag component', () => {
+  beforeEach(() => {
+    cy.server();
+    CdsHelper.setUpMocks();
+  });
+  it('Should wait for a user to accept consent before sending an event', () => {
+    Navigation.visitHomePage();
+    cy.wait(1000);
+    cy.wait(1).then(() => {
+      expect(Navigation.requestsCount(CdsHelper.clickstreamevents)).to.eq(0);
+    });
+    clickAllowAllFromBanner();
+    cy.wait(`@${CdsHelper.clickstreamevents}`).then(_ => {
+      expect(Navigation.requestsCount(CdsHelper.clickstreamevents)).to.eq(1);
+    });
+  });
+  it('Should send an event on a pageview', () => {
+    Navigation.visitHomePage();
+    clickAllowAllFromBanner();
+    cy.wait(`@${CdsHelper.clickstreamevents}`).then(xhr => {
+      ProfileTagHelper.assertPageViewEvent(xhr);
+    });
+  });
+  it('Should send an event on a productview', () => {
+    Navigation.visitHomePage();
+    clickAllowAllFromBanner();
+    cy.wait(500);
```

Comment:
same case here, this is waiting for a request to be finished I believe. 
Instead of manually waiting, you can wait for a request

Reply:
The issue here is that I want to wait for the request, but the first request has a different payload than the 2nd, and I want to assert that the first has a pageview payload and the 2nd has a productview payload. I could decrease the wait as 500 seems to be too much, but if I remove it, then it seems that the navigation to the next item happens slightly too quickly to garuntee that the pageview is sent before the productview. Does this make sense?

Reply:
we can use the cy.route to wait for that specific request, and check if it consist of the first payload and wait another wait to check for the second payload. 

Which exact request is this?